test/integration/react-18/test/concurrent.js

```diff
@@ -0,0 +1,58 @@
+/* eslint-env jest */
+
+import { join } from 'path'
+import webdriver from 'next-webdriver'
+import cheerio from 'cheerio'
+import { File, check } from 'next-test-utils'
+
+export default (context, render) => {
+  const dynamicHello = new File(
+    join(context.appDir, 'components/dynamic-hello.js')
+  )
+  const nextConfig = new File(join(context.appDir, 'next.config.js'))
+
+  async function get$(path, query) {
+    const html = await render(path, query)
+    return cheerio.load(html)
+  }
+
+  describe('concurrentFeatures is enabled', () => {
+    beforeAll(() => {
+      dynamicHello.replace('const suspense = false', `const suspense = true`)
+      nextConfig.replace(
+        'concurrentFeatures: false',
+        'concurrentFeatures: true'
+      )
+    })
+    afterAll(() => {
+      dynamicHello.restore()
+      nextConfig.restore()
+    })
+
+    it('should render the fallback on server side if not suspended on server', async () => {
+      const $ = await get$('/suspense/no-thrown')
+      const html = $('body').html()
+      expect(html).toContain('loading')
+      expect(JSON.parse($('#__NEXT_DATA__').text()).dynamicIds).toBeUndefined()
+    })
+
+    it('should render the fallback on server side if suspended on server', async () => {
+      const $ = await get$('/suspense/thrown')
+      const html = $('body').html()
+      expect(html).toContain('loading')
+      expect(JSON.parse($('#__NEXT_DATA__').text()).dynamicIds).toBeUndefined()
```

Comment:
Hmm, this isn't actually the behavior I'd expect when using `concurrentFeatures`. It should be waiting until the suspense boundary resolves before flushing. Perhaps related to the comment about throwing different promises above?